apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,366 @@
+
+numeric_project_id = 12345;
+numeric_project_id_token = u'<NUMERIC_PROJECT_ID>';
+project_id = u'fake-project';
+project_id_token = u'<PROJECT_ID>';
+
+app = Flask(__name__)
+
+def excludeKeyInRecursiveGet(key):
+  return key == u'token'
+
+def formatResponse(data, status_code, content_type):
+  return data, status_code, {common.content_type: content_type}
+
+def formatOrderedDictContents(data):
+  def getKeyDescription(key):
+    if not excludeKeyInRecursiveGet(key):
+      if isinstance(data[key], (IndexedList, OrderedDict)):
+        return key + u'/\n'
+
+    return key + u'\n'
+
+  return ''.join(getKeyDescription(key) for key in data)
+
+class HeaderCheckMiddleware(object):
+  def __init__(self, app):
+     self.app = app
+
+  def __call__(self, environ, start_response):
+    req = Request(environ, shallow=True)
+    if (req.headers.get(common.metadata_flavor) != common.metadata_flavor_google):
+      start_response('403 Forbidden',
+                     [(common.content_type, common.content_type_html),
+                      (common.metadata_flavor, common.metadata_flavor_google),
+                      (common.server, common.server_value)])
+      return ['Missing Metadata-Flavor:Google header.']
+    return self.app(environ, start_response)
+
+# Collection whose content is accessed via an index value in the relative URL.
+class IndexedList:
+  def __init__(self, items):
+    self._items = items
+
+  def __iter__(self):
+    return self._items.__iter__()
+
+  def __len__(self):
+    return self._items.__len__()
+
+  def __delitem__(self, key):
+    self._items.__delitem__(key)
+
+  def __getitem__(self, key):
+    return self._items.__getitem__(key)
+  
+  def __setitem__(self, key, value):
+    self._items.__setitem__(key, value)
+
+  def formatContents(self):
+    return ''.join(unicode(i) + u'/\n' for i in range(len(self._items)))
+
+class MetadataJSONEncoder(json.JSONEncoder):
+  def ensureOrderedDict(self, o):
+    if isinstance(o, OrderedDict):
+      return self.fixupOrderedDict(o)
+
+    return o
+
+  def fixupOrderedDict(self, d):
+    result = OrderedDict()
+    for key in d:
+      if excludeKeyInRecursiveGet(key):
+        continue
+
+      if key.endswith(u'@developer.gserviceaccount.com'):
+        newKey = key
+      else:
+        newKey = common.convertToCamelCase(key)
+
+      if (d[key] == numeric_project_id_token):
+        result[newKey] = numeric_project_id
+      else:
+        result[newKey] = self.default(d[key])
+
+    return result
+
+  def encode(self, o):
+    return json.JSONEncoder.encode(self, self.ensureOrderedDict(o))
+
+  def default(self, o):
+    if isinstance(o, IndexedList):
+      return [self.default(item) for item in o]
+    
+    return self.ensureOrderedDict(o)
+
+not_found_error = formatResponse(u'Error: Not found', 404, common.content_type_html)
+bad_request_error = formatResponse(u'Error: bad request', 400, common.content_type_html)
+
+all_data = {}
```

Comment:
global mutable state... is there any reason to prefer this, than to have loadData return the data it loads, and then pass that data around as necessary (such as to getRawDataAt?)